src/unix/glegl.cpp

```diff
@@ -0,0 +1,646 @@
+// ----------------------------------------------------------------------------
+
+wxGLCanvasEGL::wxGLCanvasEGL()
+{
+    m_config = NULL;
+    m_display = NULL;
+    m_surface = EGL_NO_SURFACE;
+    m_wl_compositor = NULL;
+    m_wl_subcompositor = NULL;
+    m_wl_frame_callback_handler = NULL;
+    m_wl_egl_window = NULL;
+    m_wl_surface = NULL;
+    m_wl_region = NULL;
+    m_wl_subsurface = NULL;
+    m_ready_to_draw = false;
+}
+
+bool wxGLCanvasEGL::InitVisual(const wxGLAttributes& dispAttrs)
+{
+    bool ret = InitConfig(dispAttrs, &m_config);
+    if ( !ret )
+    {
+        wxFAIL_MSG("Failed to get an EGLConfig for the requested attributes.");
+    }
+    return ret;
+}
+
+/* static */
+EGLDisplay wxGLCanvasEGL::GetDisplay()
+{
+    wxDisplayType type = wxDisplayNone;
+    void *display = wxGetDisplayAndType(&type);
+    EGLenum platform;
+    if (type == wxDisplayX11)
+        platform = EGL_PLATFORM_X11_EXT;
+    else if (type == wxDisplayWayland)
+        platform = EGL_PLATFORM_WAYLAND_EXT;
+    else
+        return EGL_NO_DISPLAY;
+
+    return eglGetPlatformDisplay(platform, display, NULL);
+}
+
+static void wl_global (void *data,
+                       struct wl_registry *wl_registry,
+                       uint32_t name,
+                       const char *interface,
+                       uint32_t)
+{
+    wxGLCanvasEGL *glc = (wxGLCanvasEGL *) data;
+
+    if (!strcmp (interface, "wl_compositor"))
+        glc->m_wl_compositor = (struct wl_compositor *) wl_registry_bind (wl_registry, name, &wl_compositor_interface, 3);
+    else if (!strcmp (interface, "wl_subcompositor"))
+        glc->m_wl_subcompositor = (struct wl_subcompositor *) wl_registry_bind (wl_registry, name, &wl_subcompositor_interface, 1);
+}
+
+static void wl_global_remove (void *,
+                              struct wl_registry *,
+                              uint32_t)
+{
+}
+
+static const struct wl_registry_listener wl_registry_listener = {
+    wl_global,
+    wl_global_remove
+};
+
+static void wl_frame_callback_handler(void* data,
+                                      struct wl_callback *,
+                                      uint32_t)
+{
+    wxGLCanvasEGL *glc = (wxGLCanvasEGL *) data;
+    glc->m_ready_to_draw = true;
+    g_clear_pointer(&glc->m_wl_frame_callback_handler, wl_callback_destroy);
+    glc->SendSizeEvent();
+    gtk_widget_queue_draw(glc->m_wxwindow);
+}
+
+static const struct wl_callback_listener wl_frame_listener = {
+    wl_frame_callback_handler
+};
+
+static void gtk_glcanvas_size_callback(GtkWidget *widget,
+                                       GtkAllocation *,
+                                       wxGLCanvasEGL *win)
+{
+    int scale = gtk_widget_get_scale_factor(widget);
+    wl_egl_window_resize(win->m_wl_egl_window, win->m_width * scale,
+                         win->m_height * scale, 0, 0);
+}
+
+bool wxGLCanvasEGL::CreateSurface()
+{
+    m_display = GetDisplay();
+    if ( m_display == EGL_NO_DISPLAY ) {
```

Comment:
Style nitpicking, here and below:
```suggestion
    if ( m_display == EGL_NO_DISPLAY )
    {
```